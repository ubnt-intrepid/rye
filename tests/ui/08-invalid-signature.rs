fn main() {}

#[rye::test]
fn nonunit_ret() -> std::io::Result<()> {
    Ok(())
}

#[rye::test]
fn has_input(x: u32) {}

#[rye::test]
fn has_generics<T>() {}
