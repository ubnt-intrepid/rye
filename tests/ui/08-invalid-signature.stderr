error: cannot return non-unit value from test function
 --> $DIR/08-invalid-signature.rs:4:18
  |
4 | fn nonunit_ret() -> std::io::Result<()> {
  |                  ^^^^^^^^^^^^^^^^^^^^^^

error: test functions cannot accept arguments
 --> $DIR/08-invalid-signature.rs:9:14
  |
9 | fn has_input(x: u32) {}
  |              ^^^^^^

error: test functions cannot take generic parameters
  --> $DIR/08-invalid-signature.rs:12:16
   |
12 | fn has_generics<T>() {}
   |                ^^^
